MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/location/LocationCameraController.java

```diff
@@ -355,6 +356,17 @@ private void notifyCameraTrackingChangeListener(boolean wasTracking) {
     }
   }
 
+  private MapboxMap.OnCameraMoveListener onCameraMoveListener = new MapboxMap.OnCameraMoveListener() {
+
+    @Override
+    public void onCameraMove() {
```

Comment:
Looking at https://github.com/mapbox/mapbox-gl-native-android/blob/de5003813e4bf9d88d6d9eb0980c2ec60f0006bb/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Transform.java#L112-L116

the `onCameraMove` will never get called for `moveTo` (which camera controller calls) and it needs to be fixed. If that's handled, we won't need the logic that `adjustFocalPoint` guards in this class, it would be automatically executed with `onCameraMove` listener.

Reply:
Moving `invalidateCameraPosition();` before `cameraChangeDispatcher.onCameraIdle(); ` should resolve the original issue as far as I can tell.